### PR TITLE
Adding optional clipping to tippecanoe-overzoom

### PR DESCRIPTION
**Pull request by e-n-f**
This PR adds `--clip-bounding-box` and `--clip-polygon` options to `tippecanoe-overzoom`, to clip the output tile to arbitrary rectangular or polygon bounds.

To accomplish LineString clipping to polygon bounds, it brings back [Clipper](https://www.angusj.com/clipper2/Docs/Overview.htm) as a vendored package. Wagyu continues to be used to clip and clean polygons.

### CODE REVIEW

File: overzoom.cpp [hunk 1]
```diff
@@ -67,7 +69,10 @@ int main(int argc, char **argv) {
 		{"assign-to-bins", required_argument, 0, 'b' & 0x1F},
 		{"bin-by-id-list", required_argument, 0, 'c' & 0x1F},
 		{"accumulate-numeric-attributes", required_argument, 0, 'a' & 0x1F},
+		{"numeric-attributes", required_argument, 0, 'a' & 0x1F},
```
Comment (e-n-f) [inline overzoom.cpp:72]: This looks like a merge conflict accidental edit

File: overzoom.cpp [hunk 2]
```diff
@@ -158,6 +163,25 @@ int main(int argc, char **argv) {
...
+				clipbboxes.push_back(clip);
+			} else {
+				fprintf(stderr, "%s: Can't parse bounding box --clip-bounding-box=%s\n", argv[0], optarg);
+				exit(EXIT_ARGS);
+			}
+			break;
+		}
+
+		case 'l' & 0x1F: {
+			clipbbox clip = parse_clip_poly(optarg);
```
Comment (e-n-f) [inline overzoom.cpp:180]: Clip the clip regions to the tile bounds to speed up the case where complicate bounds entirely enclose the tile.

**Review by ChrisLoer — APPROVED** [review]
I walked through this with Erica and we tested some usage (although in a pretty limited way) in Felt. Thanks for adding the clipping simplification and associated test!

**Review by ChrisLoer — APPROVED** [review]
I walked through this with Erica and we tested some usage (although in a pretty limited way) in Felt. Thanks for adding the clipping simplification and associated test!

**Pull request merged**